shared/src/commonMain/kotlin/com/br/kmpdemo/viewmodels/HomeViewModel.kt

```diff
@@ -1,55 +1,161 @@
-import moe.tlaster.precompose.viewmodel.ViewModel
-
-class HomeViewModel : ViewModel() {
-    // Home Weather Overlay
-    val location = MutableStateFlow<String?>(null)
-    val currentTemp = MutableStateFlow<String?>(null)
-    val weatherDescription = MutableStateFlow<String?>(null)
-    val tempHigh = MutableStateFlow<String?>(null)
-    val tempLow = MutableStateFlow<String?>(null)
-
-    // Forecasts
-    private val initForecasts = List(10) { ForecastState(weatherIcon = WeatherEnum.DAY_FOG) }
-    val hourlyForecasts = MutableStateFlow(initForecasts)
-    val weeklyForecasts = MutableStateFlow(initForecasts)
-
-    // Weather Details
-    private val airQualityState = MutableStateFlow(AirQualityEnum.UNKNOWN)
-    private val feelsLikeState = MutableStateFlow(FeelsLikeState())
-    private val humidityState = MutableStateFlow(HumidityState())
-    private val barometricPressureState = MutableStateFlow(BarometricPressureState())
-    private val rainFallState = MutableStateFlow(RainFallState())
-    private val sunriseSunsetState = MutableStateFlow(SunriseSunsetState())
-    private val uvIndexState = MutableStateFlow(UVIndexEnum.UNKNOWN)
-    private val visibilityState = MutableStateFlow(VisibilityState())
-    private val windState = MutableStateFlow(WindState())
-
-    val weatherDetailsState = MutableStateFlow(
-        WeatherDetailsState(
-            airQuality = airQualityState.value,
-            feelsLikeState = feelsLikeState.value,
-            humidityState = humidityState.value,
-            pressureState = barometricPressureState.value,
-            rainFallState = rainFallState.value,
-            sunriseSunsetState = sunriseSunsetState.value,
-            uvIndexState = uvIndexState.value,
-            visibilityState = visibilityState.value,
-            windState = windState.value
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import moe.tlaster.precompose.viewmodel.viewModelScope
+import org.koin.core.component.inject
+
+class HomeViewModel : BaseViewModel() {
+    private val weatherRepo: WeatherRepository by inject()
+
+    /** Forecast Responses */
+    val initForecasts = List(10) { ForecastState(weatherIcon = WeatherEnum.SUNNY) }
+    private val hourlyResponse = MutableStateFlow<Forecast?>(null)
+    private val dailyResponse = MutableStateFlow<Forecast?>(null)
+    val realTimeResponse = MutableStateFlow<RealTime?>(null)
+
+    /** UI Forecasts */
+    val hourlyForecasts = hourlyResponse
+        .map { it?.timelines?.hourly?.toHourlyForecastState() ?: initForecasts }
+        .stateIn(viewModelScope, SharingStarted.Lazily, null)
+    val dailyForecasts = dailyResponse
+        .map { it?.timelines?.daily?.toDailyForecastState() ?: initForecasts }
+        .stateIn(viewModelScope, SharingStarted.Lazily, null)
```

Comment:
Could either create an extension function for the stateIn call, or handle default state in the converter function.